### PR TITLE
chore: drop Node < 18 and use @eslint/js v9 in eslint-config-eslint

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

Updates `eslint-config-eslint` to use `@eslint/js` v9. Also updates its `package.engines` accordingly.

#### What changes did you make? (Give an overview)

Updated eslint-config-eslint as per the above.

### CODE REVIEW

File: packages/eslint-config-eslint/base.js [hunk 1]
```diff
@@ -68,7 +68,6 @@ const jsConfigs = [js.configs.recommended, {
         "no-caller": "error",
         "no-confusing-arrow": "error",
         "no-console": "error",
-        "no-constant-binary-expression": "error",
```
Comment (mdjermanovic) [inline packages/eslint-config-eslint/base.js:71]: This rule is now in the `recommended` config so no need to enable it here again.

**Review by aladdin-add — CHANGES_REQUESTED** [review]
we could also replace `no-new-object` => `no-object-constructor` (it has been deprecated in eslint v8.50.0). otherwise LGTM.

**Comment by mdjermanovic** [comment]
Good catch! Fixed in https://github.com/eslint/eslint/pull/18323/commits/31129e17ba6fded393e498ce8304140c8db460b4

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks!

**Pull request merged**